src/diepvries/table.py

```diff
@@ -2,18 +2,78 @@
 
 import logging
 from abc import ABC, abstractmethod
+from datetime import datetime
 from functools import cached_property
 from typing import Dict, List
 
 from . import HASH_DELIMITER, METADATA_FIELDS, FieldRole, FixedPrefixLoggerAdapter
 from .field import Field
-from .template_sql.sql_formulas import HASHKEY_SQL_TEMPLATE
+from .template_sql.sql_formulas import (
+    HASHKEY_SQL_TEMPLATE,
+    STAGING_PHYSICAL_NAME_SQL_TEMPLATE,
+)
 
 
 class Table(ABC):
+    """A generic table.
+
+    Abstract class Table. Holds common properties between all database tables.
+    """
+
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self, schema: str, name: str, *_args, **_kwargs):
+        """Instantiate a table.
+
+        Args:
+            schema: Data Vault schema name.
+            name: Data Vault table name.
+            _args: Unused, useful for child classes.
+            _kwargs: Unused, useful for child classes.
+        """
+        self.schema = schema
+        self.name = name.lower()
+
+        self._logger = FixedPrefixLoggerAdapter(logging.getLogger(__name__), str(self))
+
+        self._logger.info("Instance of (%s) created", type(self))
+
+    def __str__(self) -> str:
+        """Representation of a Table object as a string.
+
+        This helps the tracking of logging events per entity.
+
+        Returns:
+            String representation of a Table.
+        """
+        return f"{type(self).__name__}: {self.schema}.{self.name}"
+
+
+class StagingTable(Table):
+    """A table used for staging."""
+
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self, schema: str, name: str, extract_start_timestamp: datetime):
+        """Instantiate a StagingTable.
+
+        Args:
+             schema: Schema name.
+             name: Table name.
+             extract_start_timestamp: Extract start timestamp.
+        """
+        physical_name = STAGING_PHYSICAL_NAME_SQL_TEMPLATE.format(
+            staging_table=name,
+            staging_table_suffix=extract_start_timestamp.strftime("%Y%m%d_%H%M%S"),
+        )
```

Comment:
While there, I'd be up for getting rid of `STAGING_PHYSICAL_NAME_SQL_TEMPLATE` and moving that string creation here. It's not used anywhere else

Reply:
Good idea 👍